model/project_ref.go

```diff
@@ -160,45 +160,43 @@ var defaultGitHubTokenPermissionGroup = GitHubDynamicTokenPermissionGroup{
 	AllPermissions: true,
 }
 
+// noPermissionsGitHubTokenPermissionGroup is an empty, no permissions, group.
+var noPermissionsGitHubTokenPermissionGroup = GitHubDynamicTokenPermissionGroup{
+	Name:           "No Permissions",
+	AllPermissions: false,
+}
+
 // GetGitHubPermissionGroup returns the GitHubDynamicTokenPermissionGroup for the given requester.
 // If the requester is not found, it returns the default.
 func (p *ProjectRef) GetGitHubPermissionGroup(requester string) GitHubDynamicTokenPermissionGroup {
 	if p.GitHubPermissionGroupByRequester == nil {
 		return defaultGitHubTokenPermissionGroup
 	}
-	if groupName, ok := p.GitHubPermissionGroupByRequester[requester]; ok {
-		for _, group := range p.GitHubDynamicTokenPermissionGroups {
-			if group.Name == groupName {
-				return group
-			}
+	groupName, ok := p.GitHubPermissionGroupByRequester[requester]
+	if !ok {
+		return defaultGitHubTokenPermissionGroup
+	}
+	if groupName == noPermissionsGitHubTokenPermissionGroup.Name {
+		return noPermissionsGitHubTokenPermissionGroup
+	}
+	for _, group := range p.GitHubDynamicTokenPermissionGroups {
+		if group.Name == groupName {
+			return group
 		}
 	}
 	return defaultGitHubTokenPermissionGroup
 }
 
 func (p *ProjectRef) ValidateGitHubPermissionGroups() error {
 	catcher := grip.NewBasicCatcher()
-	// Group validation
 	for _, group := range p.GitHubDynamicTokenPermissionGroups {
-		if group.Name == "" {
-			catcher.Add(errors.New("group name cannot be empty"))
-		}
+		catcher.ErrorfWhen(group.Name == "", "group name cannot be empty")
 	}
-	// Requester validation
 	for requester, groupName := range p.GitHubPermissionGroupByRequester {
-		if !utility.StringSliceContains(evergreen.AllRequesterTypes, requester) {
-			catcher.Add(errors.Errorf("requester '%s' is not a valid requester", requester))
-		}
-		foundGroup := false
-		for _, group := range p.GitHubDynamicTokenPermissionGroups {
-			if groupName == group.Name {
-				foundGroup = true
-				break
-			}
-		}
-		if !foundGroup {
-			catcher.Add(errors.Errorf("group '%s' for requester '%s' not found", groupName, requester))
-		}
+		catcher.ErrorfWhen(!utility.StringSliceContains(evergreen.AllRequesterTypes, requester),
+			"requester '%s' is not a valid requester", requester)
+		catcher.ErrorfWhen(p.GetGitHubPermissionGroup(requester) == defaultGitHubTokenPermissionGroup,
```

Comment:
Why is this an error? It seems like it should be correct to use the default github token permission group? If it's not then it makes more sense for Get to return an error rather than this, but I don't understand the logic here.

Reply:
If the requester is using the default group, it will not be present in p.GitHubPermissionGroupByRequester.

This will be triggered when a requester points to a group that does not exist, and it 'defaults' to the default group.

Reply:
I can add a comment like:

```
// If the requester points a non-existent permission group, the default is returned.
// Requesters should only be set to existing permission groups.
```

if you think it's needed!